src/main/fc/board_info.c

```diff
@@ -0,0 +1,94 @@
+/*
+ * This file is part of Cleanflight and Betaflight.
+ *
+ * Cleanflight and Betaflight are free software. You can redistribute
+ * this software and/or modify this software under the terms of the
+ * GNU General Public License as published by the Free Software
+ * Foundation, either version 3 of the License, or (at your option)
+ * any later version.
+ *
+ * Cleanflight and Betaflight are distributed in the hope that they
+ * will be useful, but WITHOUT ANY WARRANTY; without even the implied
+ * warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this software.
+ *
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <stdbool.h>
+#include <string.h>
+
+#include "platform.h"
+
+#include "pg/board.h"
+
+static bool boardInformationSet = false;
+static char manufacturerId[MAX_MANUFACTURER_ID_LENGTH + 1];
+static char boardName[MAX_BOARD_NAME_LENGTH + 1];
+
+void initBoardInformation(void)
+{
+    boardInformationSet = boardConfig()->boardInformationSet;
+    if (boardInformationSet) {
+        strncpy(manufacturerId, boardConfig()->manufacturerId, MAX_MANUFACTURER_ID_LENGTH);
+        strncpy(boardName, boardConfig()->boardName, MAX_BOARD_NAME_LENGTH);
+    } else {
+        strcpy(manufacturerId, "");
```

Comment:
Will do the same as setting `[0] = '\0'`

Reply:
I know. Is there a benefit to not using the string functions?

Reply:
Direct assignment will likely result in less instructions generated

Reply:
Fair enough, will change it. I need to rebase this anyway once #5988 has been merged.

Reply:
don't underestimate gcc ... I'm almost certain that this is inlined as single byte store ..

and strcpy is easier to read